pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -128,12 +128,24 @@ func (mixerplugin) OnOutboundListener(in *plugin.InputParams, mutable *networkin
 
 	switch in.ListenerProtocol {
 	case networking.ListenerProtocolHTTP:
-		if skipHTTPFilter {
-			return nil
-		}
 		httpFilter := buildOutboundHTTPFilter(in.Push.Mesh, attrs, in.Node)
 		for cnum := range mutable.FilterChains {
-			mutable.FilterChains[cnum].HTTP = append(mutable.FilterChains[cnum].HTTP, httpFilter)
+			switch mutable.FilterChains[cnum].ListenerProtocol {
+			case networking.ListenerProtocolHTTP:
+				if !skipHTTPFilter {
+					mutable.FilterChains[cnum].HTTP = append(mutable.FilterChains[cnum].HTTP, httpFilter)
+				}
+			case networking.ListenerProtocolTCP:
```

Comment:
Is this when “use_downstream_protocol” is used ?
So previously we were just adding and http filter and now it will be outbound tcp filter?

Also why do we need to update mixer? We should not do it unless a customer is actually asking for it.

Reply:
> Is this when “use_downstream_protocol” is used ?

Yes, recently we added support for allowing other protocols at the same port where HTTP is defined via FCM. This ensures that there's feature parity for BlackHole/Passthrough.

> So previously we were just adding and http filter and now it will be outbound tcp filter?
> 

Yes, There are two filter chains, one with this match clause
```json
"filterChainMatch": {
                    "applicationProtocols": [
                        "http/1.0",
                        "http/1.1",
                        "h2c"
                    ]
                },
```
This handles HTTP and there's a additional filter without any match that's TCP Passthrough or BlackHole.

> Also why do we need to update mixer? We should not do it unless a customer is actually asking for it.

The change is minor and I just wanted to minimize surprises in 1.6. Once we remove the mixer code in 1.7, it wouldn't matter.